internal/catalogmetadata/filter/successors.go

```diff
@@ -28,7 +22,7 @@ func SuccessorsOf(installedBundle ocv1.BundleMetadata, channels ...declcfg.Chann
 		return nil, fmt.Errorf("parsing installed version constraint %q: %w", installedBundleVersion.String(), err)
 	}
 
-	successorsPredicate, err := successors(installedBundle, channels...)
+	successorsPredicate, err := legacySuccessor(installedBundle, channels...)
```

Comment:
Hi @joelanford 

Any reason why we should not remove the unused funcs raised by the lint?

Reply:
Nope! Pushed an amended commit to fix.